chain/src/txhashset/txhashset.rs

```diff
@@ -864,7 +864,7 @@ impl<'a> Committed for Extension<'a> {
 
 	fn kernels_committed(&self) -> Vec<Commitment> {
 		let mut commitments = vec![];
-		for n in 1..self.kernel_pmmr.unpruned_size() + 1 {
+		for n in 1..=self.kernel_pmmr.unpruned_size() {
```

Comment:
According to a discussion on Reddit [here](https://www.reddit.com/r/rust/comments/eiwhkn/why_is_iterator_so_much_faster/fctvmt4/), for loops with inclusive ranges might actually be significantly slower. So maybe we want to keep it like this?

(Same comment for the other spots this appears)

Reply:
Wow nice find!